pkg/didcomm/protocol/outofbandv2/service.go

```diff
@@ -242,76 +246,46 @@ func (s *Service) AcceptInvitation(i *Invitation) (string, error) { // nolint: g
 	logger.Debugf("oob/2.0 request body or Goal code is empty, oobv2.AcceptInvitation() is done but no" +
 		"target service triggered, generating a new peer DID for the first valid attachment and return it")
 
-	for _, atchmnt := range i.Requests {
-		serviceRequest, err := atchmnt.Data.Fetch()
-		if err != nil {
-			logger.Debugf("oob/2.0 fetching attachment request failed:%v, skipping attachment entry..", err)
-
-			continue
-		}
-
-		didCommMsgRequest := service.DIDCommMsgMap{}
-
-		err = didCommMsgRequest.UnmarshalJSON(serviceRequest)
-		if err != nil {
-			logger.Debugf("oob/2.0 fetching attachment request failed: %v, skipping attachment entry..", err)
-
-			continue
-		}
-
-		senderDID, ok := didCommMsgRequest["from"].(string)
-		if !ok {
-			logger.Debugf("oob/2.0 fetching attachment request does not have from field, skipping " +
-				"attachment entry..")
-
-			continue
-		}
-
-		senderDoc, err := s.vdrRegistry.Resolve(senderDID)
-		if err != nil {
-			return "", fmt.Errorf("failed to resolve inviter DID: %w", err)
-		}
-
-		myDID, err := s.vdrRegistry.Create(peer.DIDMethod, newDID)
-		if err != nil {
-			logger.Debugf("oob/2.0 creating new DID via VDR failed: %v, skipping attachment entry..", err)
-
-			continue
-		}
+	senderDoc, err := s.vdrRegistry.Resolve(i.From)
+	if err != nil {
+		return "", fmt.Errorf("oob/2.0 failed to resolve inviter DID: %w", err)
+	}
 
-		destination, err := service.CreateDestination(senderDoc.DIDDocument)
-		if err != nil {
-			return "", fmt.Errorf("failed to create destination: %w", err)
-		}
+	myDID, err := s.vdrRegistry.Create(peer.DIDMethod, newDID)
+	if err != nil {
+		return "", fmt.Errorf("oob/2.0 creating new DID via VDR failed: %w", err)
+	}
 
-		connRecord := &connection.Record{
-			ConnectionID:      uuid.New().String(),
-			ParentThreadID:    i.ID,
-			State:             "null",
-			InvitationID:      i.ID,
-			ServiceEndPoint:   destination.ServiceEndpoint,
-			RecipientKeys:     destination.RecipientKeys,
-			RoutingKeys:       destination.RoutingKeys,
-			TheirLabel:        i.Label,
-			TheirDID:          senderDID,
-			Namespace:         "my",
-			MediaTypeProfiles: s.myMediaTypeProfiles,
-			Implicit:          true,
-			InvitationDID:     myDID.DIDDocument.ID,
-			DIDCommVersion:    service.V2,
-		}
+	destination, err := service.CreateDestination(senderDoc.DIDDocument)
+	if err != nil {
+		return "", fmt.Errorf("oob/2.0 failed to create destination: %w", err)
+	}
 
-		if err := s.connectionRecorder.SaveConnectionRecord(connRecord); err != nil {
-			return "", err
-		}
+	connRecord := &connection.Record{
+		ConnectionID:      uuid.New().String(),
+		ParentThreadID:    i.ID,
+		State:             connection.StateNameCompleted,
+		InvitationID:      i.ID,
+		ServiceEndPoint:   destination.ServiceEndpoint,
+		RecipientKeys:     destination.RecipientKeys,
+		RoutingKeys:       destination.RoutingKeys,
+		TheirLabel:        i.Label,
+		TheirDID:          i.From,
```

Comment:
ah wait, don't we also need to set MyDID in the record?

Reply:
technically it's the same as `InvitationDID` further below... but i've added MyDID for the sake of completeness.

Reply:
:+1: yeah it's a bit of cruft from the record being used in different ways in didexchange vs elsewhere
but `MyDID` is needed in a lot of places, while `InvitationDID` is mostly ignored.